sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -1971,4 +1971,10 @@ object QueryExecutionErrors extends QueryErrorsBase {
         s"add ${toSQLValue(amount, IntegerType)} $unit to " +
         s"${toSQLValue(DateTimeUtils.microsToInstant(micros), TimestampType)}"))
   }
+
+  def multipleRowSubqueryError(plan: String): Throwable = {
+    new SparkIllegalStateException(
```

Comment:
Please, use SparkException or SparkRuntimeException. I removed the the exception by https://github.com/apache/spark/pull/36550.

BTW, we shouldn't throw `IllegalStateException` if the exception can be trigger by user code in regular cases.